src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -40,21 +64,209 @@ public SnapshotBuilderGetConceptsResponse getConceptChildren(UUID datasetId, Int
                     .id(conceptId + 1)));
   }
 
-  private int getRollupCount(UUID datasetId, List<SnapshotBuilderCohort> cohorts) {
-    return 100;
-  }
-
   public SnapshotBuilderCountResponse getCountResponse(
       UUID id, List<SnapshotBuilderCohort> cohorts) {
     return new SnapshotBuilderCountResponse()
         .sql("")
-        .result(new SnapshotBuilderCountResponseResult().total(getRollupCount(id, cohorts)));
+        .result(
+            new SnapshotBuilderCountResponseResult()
+                .total(
+                    getRollupCountForCriteriaGroups(
+                        id,
+                        cohorts.stream()
+                            .map(SnapshotBuilderCohort::getCriteriaGroups)
+                            .collect(Collectors.toList()))));
   }
 
   public EnumerateSnapshotAccessRequest enumerateByDatasetId(UUID id) {
     return convertToEnumerateModel(snapshotRequestDao.enumerateByDatasetId(id));
   }
 
+  public int getRollupCountForCriteriaGroups(
+      UUID datasetId, List<List<SnapshotBuilderCriteriaGroup>> criteriaGroupsList) {
+    Query query = generateRollupCountsQueryForCriteriaGroupsList(criteriaGroupsList);
+    query.renderSQL();
+    return 5;
+  }
+
+  @NotNull
+  private Query generateRollupCountsQueryForCriteriaGroupsList(
+      List<List<SnapshotBuilderCriteriaGroup>> criteriaGroupsList) {
+    TablePointer tablePointer = TablePointer.fromTableName("person");
+    TableVariable tableVariable = TableVariable.forPrimary(tablePointer);
+
+    FieldVariable personId = makePersonCountVariable();
+    return new Query(
+        List.of(personId),
+        List.of(tableVariable),
+        new BooleanAndOrFilterVariable(
```

Comment:
Question on organization - As we're adding more and more of this type of code, what would you think about moving the actual query building to another location, maybe a util?

Reply:
I agree. I also think if we had tests or java doc showing the expected sql that these render to that would help a lot with making the query builder code more readable